### PR TITLE
Make PathOrdering respect compare contract

### PR DESCRIPTION
**Pull request by TheElan**
trying to solve: https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/8770

### CODE REVIEW

**Comment by TheElan** [comment]
no luck in pinpointing problematic edge case or solving the issue so far

**Comment by TheElan** [comment]
I've not had any substantial experience with Scala before and my code might not be "up to standard". I'll gladly accept any suggestions on how to write it right scala-way

File: src/mrtjp/projectred/transportation/requests.scala [hunk 1]
```diff
@@ -376,64 +375,58 @@ class RequestRoot(thePackage:ItemKeyStack, equality:ItemEquality, requester:IWor
...
-    {
-        var x = x1
-        var y = y1
+class PathOrdering(distanceWeight: Double) extends Ordering[StartEndPath] {
+    override def compare(first: StartEndPath, second: StartEndPath): Int = {
+        val firstRouter = first.end.getParent
+        val secondRouter = second.end.getParent
 
-        var c = 0.0D
+        val priorityComparison: Int = compareByPriorities(firstRouter, secondRouter)
```
Comment (repo-alt) [inline src/mrtjp/projectred/transportation/requests.scala:390]: Explicit types for values are unusual

**Review by repo-alt — APPROVED** [review]
I can't say I understand why it was broken, but new version is definitely more readable

**Pull request merged**